opm/common/utility/OpmInputError.hpp

```diff
@@ -83,20 +83,22 @@ class OpmInputError : public std::exception {
         return this->m_what.c_str();
     }
 
-
-    static std::string format(const std::string& msg_fmt, const KeywordLocation& loc) {
-        return fmt::format(msg_fmt,
-                           fmt::arg("keyword", loc.keyword),
-                           fmt::arg("file", loc.filename),
-                           fmt::arg("line", loc.lineno));
+    template<class ... Args>
+    static std::string format(const std::string& msg_format, const KeywordLocation& loc, const Args ... additionalArguments) {
+        return fmt::format(msg_format,
+            std::forward<const Args>(additionalArguments)...,
```

Comment:
Looks good! If possible I would prefer having the extra arguments as the last arguments to the `fmt::format()` call?

Reply:
This was initially considered and tested, but rejected:

When the extra arguments are last, positional arguments supplied as extra arguments behave surprisingly.

Consider the natural thing to write:

```c++
Opm::OpmInputError::format("{keyword}@{file}:{line}: {}", location, e.what());
```

In this case, you should expect the positional parameter you apply to fill the last placeholder, as its the first placeholder for a positional argument; if the named parameters are supplied first, that placeholder would have to be manually indicated as `{3}`, essentially making its positional nature pointless.

This does mean that in general terms, strings sent to format need to explicitly use the named placeholders (if they are sending in additional parameters, anyway), but that's a far more reasonable requirement, and hides the implementation detail that is the internal ordering of the three keywords.

Furthermore, requiring the named parameters means that not only is the order no longer public, if a fourth named parameter were to be added, it would not require all the uses of `{3}` across the code to be updated to `{4}`.

Reply:
I did not understand all the the details - but the test code looks good!